app/src/test/java/waiter/MessengerTest.java

```diff
@@ -5,17 +5,40 @@
 import waiter.Messenger.Messenger;
 
 import org.junit.jupiter.api.Test;
-import static org.junit.jupiter.api.Assertions.assertEquals;
+import org.junit.jupiter.api.BeforeEach;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
 
 import java.io.IOException;
+import java.util.Arrays;
 
 class MessengerTest {
+
+    String clientHasDisconnected;
+
+    @BeforeEach
+    void setUp() {
+        clientHasDisconnected = null;
+    }
+
+    @Test
+    void echosSingleMessage() throws IOException {
```

Comment:
Can we also apply similar formatting to this test? Personally I like to group test setup into `input`, `exercise the function under test` and `assert`, each session with new lines separating them. This makes the test easier to reason about.

Reply:
Absolutely, I completely forgot this general test format, something I should definitely remember because it is good to do always!

Reply:
Addressed in [this commit](https://github.com/jonmanahan/waiter/pull/7/commits/713d471c9fe377388299139e90627d4683daca6c), formatted to have the input -> exercise -> assert clearly separated 👍